File: CJY/BJ/No_1157.js
const rl = require("readline").createInterface({input: process.stdin, output: process.stdout})

rl.on("line", (s) => {
    s = s.toUpperCase();
    let arr = Array.from({length: 26}, () => 0);
    for (let i = 0, len = s.length; i < len; i++) arr[s.charCodeAt(i) - 65]++;

    let max = 0;
    let answer;
    let b;

    for (let i = 0; i < 26; i++) {
        if(arr[i] === 0) continue;
        if(max < arr[i]) {
            max = arr[i];
            answer = String.fromCharCode(i + 65);
            b = false;
        } else if(max === arr[i]) {
            b = true;
        }
    }
    console.log(b ? '?' : answer);
    rl.close();
}).on("close", () => {process.exit();})

File: CJY/BJ/No_1157.java
import java.io.*;

public class No_1157 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String s = br.readLine().toUpperCase();
        br.close();

        int[] alp = new int[26];
        for (int i = 0, len = s.length(); i < len; i++) alp[s.charAt(i) - 65]++;

        int max = 0;
        char answer = 0;
        boolean b = false;

        for (int i = 0; i < 26; i++) {
            if(alp[i] == 0) continue;
            if(max < alp[i]) {
                max = alp[i];
                answer = (char) (i + 65);
                b = false;
            } else if(max == alp[i]) {
                b = true;
            }
        }

        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        bw.write(b ? "?" : String.valueOf(answer));
        bw.flush();
        bw.close();
    }
}
